src/test/java/uk/gov/pay/products/matchers/PaymentEntityMatcher.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.pay.products.matchers;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import uk.gov.pay.products.persistence.entity.PaymentEntity;
+
+public class PaymentEntityMatcher {
+    public static Matcher<PaymentEntity> isSame(final PaymentEntity expectedPaymentEntity) {
+        return new BaseMatcher<PaymentEntity>() {
+            @Override
+            public boolean matches(final Object obj) {
+                final PaymentEntity actualPaymentEntity = (PaymentEntity) obj;
+
+                return ((actualPaymentEntity != null) &&
+                        (expectedPaymentEntity != null) &&
+                        (actualPaymentEntity.getExternalId() != null) &&
+                        (actualPaymentEntity.getNextUrl() == expectedPaymentEntity.getNextUrl()) &&
```

Comment:
Shouldn't be `equals` instead of `==` for String equality?
Same for line `20`

Reply:
Absolutely!  Newbies mistake :-)